### PR TITLE
[#7228, #7225][Platform UI] Enable the OpenShift tab

### LINKED ISSUE #7225: [openshift] [Platform] Prefill YB DB image repo from redhat registry instead of quay.io

### LINKED ISSUE #7228: [Platform] [openshift] Enable provider tab for openshift
As part of the UI changes, we can also remove the service account field from all kubernetes providers.

### PR DESCRIPTION
**Pull request by bhavin192**
This enables the Red Hat OpenShift tab in the Configs section. The
image registry for this provider is set to
registry.connect.redhat.com/yugabytedb/yugabyte.

**_Marking the provider as alpha._**
This is how it looks without any provider:
image

With one provider:
image

Fixes #7228, fixes #7225

Scenarios tested:
- Created OpenShift, Tanzu, Kubernetes providers as follows
  - Without specifying any registry.
  - With a custom container registry.
- All these providers had correct default registries.
- These 6 providers were listed correctly under their own tabs.

### CODE REVIEW

File: managed/ui/src/components/config/PublicCloud/Kubernetes/CreateKubernetesConfiguration.js [hunk 1]
```diff
@@ -270,7 +276,8 @@ class CreateKubernetesConfiguration extends Component {
                         <Col lg={7}>
                           <Field
                             name="imageRegistry"
-                            placeholder="quay.io/yugabyte/yugabyte"
+                            placeholder={ providerTypeOptions.length === 1 && providerTypeOptions[0].value === 'openshift'
+                                          ? "registry.connect.redhat.com/yugabytedb/yugabyte" : "quay.io/yugabyte/yugabyte" }
```
Comment (sshev) [inline managed/ui/src/components/config/PublicCloud/Kubernetes/CreateKubernetesConfiguration.js:280]: let's put these urls into a `const` and then reuse it

File: managed/ui/src/components/config/PublicCloud/Kubernetes/KubernetesProviderConfiguration.js [hunk 2]
```diff
@@ -48,11 +48,13 @@ class KubernetesProviderConfiguration extends Component {
       .map((region) => {
         const providerData = providers.data.find((p) => p.uuid === region.provider.uuid);
 
-        // If the type is Tanzu we don't want to include other k8s configs and vice versa.
+        // If the type has a dedicated tab, we don't want to include
+        // other k8s configs and vice versa.
+        let dedicatedK8sTabs = ['tanzu', 'openshift']
```
Comment (sshev) [inline managed/ui/src/components/config/PublicCloud/Kubernetes/KubernetesProviderConfiguration.js:53]:
```suggestion
        const dedicatedK8sTabs = ['tanzu', 'openshift'];
```

**Review by sshev — COMMENTED** [review]
Address few minor comments and attach a full screenshot pls

File: managed/ui/src/components/config/PublicCloud/Kubernetes/CreateKubernetesConfiguration.js (hunk 1 above)
Comment (bhavin192) [reply managed/ui/src/components/config/PublicCloud/Kubernetes/CreateKubernetesConfiguration.js]: I have added them as two variables, what do you think?

Comment (bhavin192) [reply managed/ui/src/components/config/PublicCloud/Kubernetes/CreateKubernetesConfiguration.js]: Or should I add it something like:
```js
const containerRegistries = { quay: 'quay.io/yugabytedb/yugabyte', redhat: 'registry.connect.redhat.com/yugabytedb/yugabyte' };
```

**Comment by bhavin192** [comment]
@sshev thank you for a quick review. I have added full screenshots to the body of the PR.

**Review by sshev — APPROVED** [review]
LGTM

**Pull request merged**